code/modules/power/power.dm

```diff
@@ -275,54 +275,64 @@
 
 	return net1
 
-//Determines how strong could be shock, deals damage to mob, uses power.
-//M is a mob who touched wire/whatever
-//power_source is a source of electricity, can be power cell, area, apc, cable, powernet or null
-//source is an object caused electrocuting (airlock, grille, etc)
-//siemens_coeff - layman's terms, conductivity
-//dist_check - set to only shock mobs within 1 of source (vendors, airlocks, etc.)
-//No animations will be performed by this proc.
-/proc/electrocute_mob(mob/living/carbon/M, power_source, obj/source, siemens_coeff = 1, dist_check = FALSE)
-	if(!istype(M) || ismecha(M.loc))
-		return 0	//feckin mechs are dumb
-	if(dist_check)
-		if(!in_range(source,M))
-			return 0
-	if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-		if(H.gloves)
-			var/obj/item/clothing/gloves/G = H.gloves
-			if(G.siemens_coefficient == 0)
-				SEND_SIGNAL(M, COMSIG_LIVING_SHOCK_PREVENTED, power_source, source, siemens_coeff, dist_check)
-				return 0		//to avoid spamming with insulated glvoes on
-
+/// Extracts the powernet and cell of the provided power source
+/proc/get_powernet_info_from_source(power_source)
 	var/area/source_area
-	if(istype(power_source, /area))
+	if (istype(power_source, /area))
 		source_area = power_source
 		power_source = source_area.get_apc()
-	if(istype(power_source, /obj/structure/cable))
+	else if (istype(power_source, /obj/structure/cable))
 		var/obj/structure/cable/Cable = power_source
 		power_source = Cable.powernet
 
 	var/datum/powernet/PN
 	var/obj/item/stock_parts/cell/cell
 
-	if(istype(power_source, /datum/powernet))
+	if (istype(power_source, /datum/powernet))
 		PN = power_source
-	else if(istype(power_source, /obj/item/stock_parts/cell))
+	else if (istype(power_source, /obj/item/stock_parts/cell))
 		cell = power_source
-	else if(istype(power_source, /obj/machinery/power/apc))
+	else if (istype(power_source, /obj/machinery/power/apc))
 		var/obj/machinery/power/apc/apc = power_source
 		cell = apc.cell
 		if (apc.terminal)
 			PN = apc.terminal.powernet
 	else if (!power_source)
-		return 0
+		return
 	else
-		log_admin("ERROR: /proc/electrocute_mob([M], [power_source], [source]): wrong power_source")
-		return 0
+		return
```

Comment:
Looks like the `else if (!power_source)` check has become redundant, as the proc will return regardless of it being true or false?